recipes/extra-cmake-modules/all/test_package/conanfile.py

```diff
@@ -1,17 +1,34 @@
-from conans import ConanFile, CMake, tools
+from conan import ConanFile
+from conan.tools.cmake import CMake, CMakeDeps, CMakeToolchain, cmake_layout
+from conan.tools.build import can_run
+from conan.tools.microsoft import is_msvc
 import os
 
 class ExtraCMakeModulesTestConan(ConanFile):
     settings = "os", "compiler", "arch", "build_type"
-    generators = "cmake"
+
+    def build_requirements(self):
+        self.requires(self.tested_reference_str)
+
+    def generate(self):
+        tc = CMakeToolchain(self)
+        tc.generate()
+        deps = CMakeDeps(self)
+        deps.generate()
+
+    def layout(self):
+        cmake_layout(self)
 
     def build(self):
         cmake = CMake(self)
         cmake.configure()
         cmake.build()
 
     def test(self):
-        if not tools.cross_building(self):
-            bin_path = os.path.join("bin", "example")
-            self.run(bin_path, run_environment=True)
+        if can_run(self):
+            progname = "example"
+            if is_msvc(self):
+                progname += ".exe"
 
+            runpath = os.path.join(self.cpp.build.bindirs[0], progname)
+            self.run(runpath, env="conanrun")
```

Comment:
```suggestion
        if can_run(self):
            bin_path = os.path.join(self.cpp.build.bindir, "example")
            self.run(bin_path, env="conanrun")
```


You could simplify even more: https://github.com/conan-io/conan-center-index/blob/master/docs/package_templates/cmake_package/all/test_package/conanfile.py

Reply:
I had a difficult time getting this to compile on Windows under MSBuild. But I've checked on my own windows machine and this now does seem to work, so have addid this change to commits below

Reply:
@weatherhead99 on Windows, is not needed to add `.exe` suffix, the cli should solve it automatically. It's not a Conan stuff, but default Windows behavior.

Reply:
fine. I only added this in the first place because it wouldn't run in my particular powershell configuration but I'm no windows expert so I assume my system is weird somehow